### DISCUSSION TITLE
Dag Instance went to success state with out executing its tasks

### DISCUSSION THREAD

**Discussion by Venkatesh748** [apache/airflow#28984] [Q&A]
Hi All,

I have two dags, one dag will trigger another dag by using TriggerDagRunOperator. 

consider below example 

DAG-A                 DAG-B
Task-A1               Task-B1
Task-A2               Task-B2

DAG A is scheduled one and DAG B is not scheduled dag.

In Task-A2 of DAG-A we are triggering DAG-B using TriggerDagRunOperator.

Here the issue once DAG-A trigger DAG-B, I see that DAG-B instance state as success with out executing its tasks. This happens randomly not for every triggering. Recently we are facing this issue frequently which is impacting our application.

trigger = TriggerDagRunOperator(
    task_id='trigger_id_<datetime>',
    trigger_dag_id='<dagname>',
    conf={"procssid":"<appropriate value>"},
    dag=dag1
    )
trigger.execute(context=context)

I am using static date for starte date in default args like below
'start_date': datetime(2020, 1, 1, 0, 0)

In airflow we have increased dag concurrency limit and tried every thing but nothing worked out.

![airflow_dag_stuck](https://user-images.githubusercontent.com/53960890/212875522-c40fd279-3a2c-4a72-a244-93e2a209d0b1.jpg)

**Comment by potiuk** [comment]
I think more logs around the issue and information which version of Airflow you have would be useful. Also upgrading to latest version of Airlfow is always a good idea  - we fix literally tens of errors with every bugfix release, so trying to upgrade to latest airlfow first is always a good idea, because sometimes it might even be difficult to guess if the problem of yours is a manifestation of another error that has been already fixed.

**Reply by Venkatesh748** [reply]
Thanks for the reply @potiuk , we are using airflow version v2.0.1. We don't see any log related to this issue.

**Reply by potiuk** [reply]
Then definitely UPGRADE. There were literally thousands of bugfixes since v2.0.1. Using it makes completely no sense, especially that there were even releases in the 2.0* line with only bugfixes.
